waiting-for-start.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+echo "waiting for servers to start..."
+declare -a arr=("npm" "java")
+for pid in ${arr[@]}; do
+  while true; do
+    echo $pid
+    if ! pgrep -n -f $pid > /dev/null; then
+      sleep 1;
+      echo "still waiting..."
+      echo $(pgrep -n -f npm)
+    else
+      echo "All needed background processes start running!"
```

Comment:
You are right. So how about having a simple HEAD request to the localhost?
To specify, selenium driver will be running on  http://localhost:4444/wd/hub
and node will be running on localhost:3000

2016-06-28 11:07 GMT-05:00 Xingchen Hong <notifications@github.com>:

> In waiting-for-start.sh
> <https://github.com/Zodiase/map-visualizer/pull/15#discussion_r68788438>:
>
> > @@ -0,0 +1,17 @@
> > +#!/bin/bash
> > +
> > +echo "waiting for servers to start..."
> > +declare -a arr=("npm" "java")
> > +for pid in ${arr[@]}; do
> > +  while true; do
> > +    echo $pid
> > +    if ! pgrep -n -f $pid > /dev/null; then
> > +      sleep 1;
> > +      echo "still waiting..."
> > +      echo $(pgrep -n -f npm)
> > +    else
> > +      echo "All needed background processes start running!"
>
> Does the appearance of process guarantee the state of the process? A
> server process might be in the process of starting and not yet ready for
> connections.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Zodiase/map-visualizer/pull/15/files/1c3d4e5ef0ec58890c6b2a5a60b21090902a2c76#r68788438>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJZCbalekXR3dVqMuSbMa7Rc1RlmybPDks5qQUbTgaJpZM4JAOXy>
> .
>